### PR TITLE
add basic test for CLI tools

### PR DESCRIPTION
**Pull request by driftregion**
This parameterized test catches https://github.com/FreeOpcUa/opcua-asyncio/issues/106

### CODE REVIEW

**Comment by oroulet** [comment]
I think there are tests for tools , but maybe not for all. so this is fine. But it looks like you need to make it pass :-)

**Comment by oroulet** [comment]
What is broken?

**Comment by swamper123** [comment]
> tests/test_tools.py::test_that_tool_can_be_invoked_without_internal_error[uasubscribe]
> 
>   /home/travis/build/FreeOpcUa/opcua-asyncio/tests/test_tools.py:15: RuntimeWarning: coroutine 'uasubscribe' was never awaited
> 
>     tool()

behind `uasubscribe ` is a Coroutine, so it crashes here. Maybe using `asycio.run()` could solve the problem, because it doesn't matter if it runs a coro or a function.

https://github.com/FreeOpcUa/opcua-asyncio/blob/3353bc874c4511dab7187bf8d7614309331cf903/asyncua/tools.py#L388

File: asyncua/tools.py [hunk 1]
```diff
@@ -614,26 +616,30 @@ async def _uadiscover():
...
-
-    for i, ep in enumerate(await client.connect_and_get_server_endpoints(), start=1):
-        print('Endpoint {0}:'.format(i))
-        for (n, v) in endpoint_to_strings(ep):
-            print('  {0}: {1}'.format(n, v))
-        print('')
+        for i, ep in enumerate(await client.connect_and_get_server_endpoints(), start=1):
+            print('Endpoint {0}:'.format(i))
+            for (n, v) in endpoint_to_strings(ep):
+                print('  {0}: {1}'.format(n, v))
```
Comment (oroulet) [inline asyncua/tools.py:638]: you can use f-string instead of format

**Comment by driftregion** [comment]
Hi, I'm having a much better development experience with https://github.com/FreeOpcUa/python-opcua.
I'm closing this to focus on issues there.

**Comment by oroulet** [comment]
@driftregion what issues do you have with opcua.-python that you donot have with python-opcua? python-opcua isin maintenance mode so I encourage everyone to focus on opcua-asyncio for new things

**Comment by driftregion** [comment]
thanks for asking

It's my workflow that's the issue.  ipdb is such a powerful tool, I rarely
need to search documentation anymore when I can drop an ipdb breakpoint in
the program and dig around until I find what I'm looking for.

last I experienced, ipdb is unpleasant to use with async libs including
asyncio.  I.e. one can break inside coroutines, but can't await other
coroutines, which makes figuring out large complex IO libraries
exceptionally difficult.

can you share what your workflow is like in this regard?

**Comment by swamper123** [comment]
@driftregion It seems that IPython is on this problem. Async/Await is supported with Version IPython 7.0 and since then it still has some features in experimental.
Anyway they are working on it in the ipython repo and with the last update they even had a fix for ipdb.

My 5 cents for the workflow question: I am using the PyCharm debugger, which is a great tool in my opinion. It shows everything about a variable, works without any trouble with async stuff and the IDE is pretty neat, too. I haven't worked so much with IPython, so I can't compare these two with one another.

**Comment by oroulet** [comment]
At least I know ipython/embed has quite good suport for coroutines. Not sure that works all the time though...  `%autowait on`

**Comment by swamper123** [comment]
I had a look on [the latest test reports e.g.](https://travis-ci.org/github/FreeOpcUa/opcua-asyncio/jobs/724503867#L741). It seems that `test_that_tool_can_be_invoked_without_internal_error` throws errors in the subprocess, but tests still pass. 
I guess this isn't wanted behaviour?

**Pull request merged**